server/channels/app/platform/license.go

```diff
@@ -199,7 +201,9 @@ func (ps *PlatformService) SaveLicense(licenseBytes []byte) (*model.License, *mo
 	sysVar.Name = model.SystemActiveLicenseId
 	sysVar.Value = license.Id
 	if err := ps.Store.System().SaveOrUpdate(sysVar); err != nil {
-		ps.RemoveLicense()
+		if err := ps.RemoveLicense(); err != nil {
```

Comment:
For consistency, we may also want to log the original error (from the `SaveOrUpdate` call) since it's essentially swallowed. Another option would be to wrap it in the returned AppError.